src/Microsoft.Diagnostics.ExtensionCommands/NativeAddressHelper.cs

```diff
@@ -16,9 +16,18 @@ namespace Microsoft.Diagnostics.ExtensionCommands
     public sealed class NativeAddressHelper
     {
         private ((bool, bool, bool, bool) Key, DescribedRegion[] Result) _previous;
+        private ITarget _target;
 
         [ServiceImport]
-        public ITarget Target { get; set; }
+        public ITarget Target
+        {
+            get => _target;
+            set
+            {
+                value?.OnFlushEvent.Register(() => _previous = default);
```

Comment:
Adding the OnFlushEvent is usually done in the service constructor and the result IDisposable needs to be saved and disposed in (new) IDisposable.Dispose() function:

Something like 
``` C#
public NativeAddressHelper(ITarget target)
{
     _onFlushEvent = target.OnFlushEvent.Register(() => _previous = default);
}

IDisposable.Dispose()
{
      _onFlushEvent.Dispose();
}
```

Reply:
The lifetime of the ITarget service the flush is registered is longer than a Runtime scoped service.